disco_aws_automation/asiaq_cli.py

```diff
@@ -0,0 +1,160 @@
+      input and adds to it whatever arguments and options this command needs.
+    """
+
+    DESCRIPTION = "This command has no description.  Perhaps you should add one?"
+
+    def __init__(self):
+        self.logger = getLogger(type(self).__name__)
+
+    def run(self, args):
+        raise Exception("This is an abstact method. Override it so this command does something useful!")
+
+    @classmethod
+    def init_args(cls, parser):
+        """
+        Set up any arguments and options for this command/subcommand.
+        """
+        pass
+
+
+class SandboxCommand(CliCommand):
+    """
+    Command to manage sandboxes (currently, just creates, but should get other functions later).
+    """
+
+    DESCRIPTION = "Create and populate a sandbox for local development and testing."
+
+    PEERING_SECTION = 'peerings'
+    BLACK_LIST_S3_CONTAINER = "zookeeper-sync-black-lists"
+    BLACK_LIST_BUCKET = 'us-west-2.mclass.sandboxes'
+
+    @classmethod
+    def init_args(cls, parser):
+        parser.add_argument("sandbox_name")
+
+    def run(self, args):
+        self.logger.debug("Updating sandbox %s", args.sandbox_name)
+        sandbox_name = args.sandbox_name
+        pipeline_file = os.path.join("sandboxes", sandbox_name, "pipeline.csv")
+
+        aws_config = read_config()
+        hostclass_dicts = read_pipeline_file(pipeline_file)
+
+        self._update_blacklist(sandbox_name)
+
+        self.logger.info("Checking if environment '%s' already exists", sandbox_name)
+        vpc = DiscoVPC.fetch_environment(environment_name=sandbox_name)
+        if vpc:
+            self.logger.info("Sandbox %s already exists: updating it.", sandbox_name)
+            vpc.update()
+        else:
+            vpc = DiscoVPC(environment_name=sandbox_name,
+                           environment_type='sandbox',
+                           defer_creation=True)
+            peering_found = False
+            peering_prefixes = ("*:sandbox", ("%s:sandbox" % sandbox_name))
+            if vpc.config.has_section(self.PEERING_SECTION):
+                for peering in vpc.config.options(self.PEERING_SECTION):
+                    peers = vpc.config.get(self.PEERING_SECTION, peering)
+                    self.logger.debug("Peering config: %s = '%s'", peering, peers)
+                    if peers.startswith(peering_prefixes):
+                        peering_found = True
+                        break
+                    elif peering.endswith("_99"):
+                        raise Exception("oh this is going to be a problem")
+            else:
+                self.logger.warn("No peering section found")
+                vpc.config.add_section(self.PEERING_SECTION)
+            if not peering_found:
+                self.logger.warn("Need to update peering config for %s", sandbox_name)
+                vpc.config.set(self.PEERING_SECTION, "connection_99",
+                               "%s:sandbox/intranet ci/intranet" % sandbox_name)
```

Comment:
This assumes the envtype for sandboxes is `sandbox`. There has been at least one case so far where a team used a different envtype for a sandbox.

Also, Asiaq doesn't really have a concept of sandboxes. Its something we invented outside of Asiaq by pulling together VPCs, peering with CI, zookeeper magic and the black list files. None of that stuff is really part of Asiaq